testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDatabricksUnityCompatibility.java

```diff
@@ -0,0 +1,166 @@
+            onDelta().executeQuery(format("INSERT INTO %s.%s values (1, 'one')", unityCatalogCatalogName, tableName));
+
+            assertThat(onTrino().executeQuery("SHOW SCHEMAS FROM delta"))
+                    .contains(ImmutableList.of(
+                            row(schemaName.toLowerCase(Locale.ROOT))));
+
+            assertThat(onTrino().executeQuery("SHOW TABLES in delta." + schemaName))
+                    .containsOnly(row("testtable"));
+
+            // select
+            assertThat(onTrino().executeQuery(format("SELECT * FROM delta.%s", tableName)))
+                    .containsOnly(row(1, "one"));
+
+            // insert
+            onTrino().executeQuery(format("INSERT INTO delta.%s values (2, 'two')", tableName));
+            assertThat(onTrino().executeQuery(format("SELECT * FROM delta.%s order by c1", tableName)))
+                    .containsOnly(row(1, "one"), row(2, "two"));
+            assertThat(onDelta().executeQuery(format("SELECT * FROM %s.%s order by c1", unityCatalogCatalogName, tableName)))
+                    .containsOnly(row(1, "one"), row(2, "two"));
+            // update
+            onTrino().executeQuery(format("UPDATE delta.%s SET c2 = 'two hundred' WHERE c1 = 2", tableName));
+            assertThat(onTrino().executeQuery(format("SELECT * FROM delta.%s order by c1", tableName)))
+                    .containsOnly(row(1, "one"), row(2, "two hundred"));
+            assertThat(onDelta().executeQuery(format("SELECT * FROM %s.%s order by c1", unityCatalogCatalogName, tableName)))
+                    .containsOnly(row(1, "one"), row(2, "two hundred"));
+
+            // delete
+            onTrino().executeQuery(format("DELETE FROM delta.%s WHERE c2 = 'one'", tableName));
+            assertThat(onTrino().executeQuery(format("SELECT * FROM delta.%s order by c1", tableName)))
+                    .containsOnly(row(2, "two hundred"));
+            assertThat(onDelta().executeQuery(format("SELECT * FROM %s.%s order by c1", unityCatalogCatalogName, tableName)))
+                    .containsOnly(row(2, "two hundred"));
+
+            // merge
+            String tableName2 = schemaName + ".testTable2";
+            String tableLocation2 = getRandomPathInExternalLocation(tableName2);
+            onDelta().executeQuery(format("CREATE TABLE %s.%s (c1 int, c2 string) using delta location 's3://%s/%s'", unityCatalogCatalogName, tableName2, bucketName, tableLocation2));
+            onDelta().executeQuery(format("INSERT INTO %s.%s values (1, 'one'), (2, 'two'), (3, 'three')", unityCatalogCatalogName, tableName2));
+
+            onTrino().executeQuery(format("MERGE INTO delta.%s t USING delta.%s s on t.c1 = s.c1 " +
+                    "WHEN MATCHED THEN UPDATE SET c2 = s.c2 " +
+                    "WHEN NOT MATCHED THEN INSERT (c1, c2) VALUES (s.c1, s.c2)", tableName, tableName2));
+            assertThat(onTrino().executeQuery(format("SELECT * FROM delta.%s order by c1", tableName)))
```

Comment:
Extract the expected rows

https://github.com/trinodb/trino/blob/470f666be5a1cec41891963f842a7f3e6a808f1d/testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java#L71-L75